### ISSUE TITLE
"System cannot find the file specified" when attempting to update Domain Shared contact photo via "gam update contactphotos"

### ISSUE THREAD

**Issue by gkmachine** [taers232c/GAMADV-XTD3#255]
When attempting to use GAMADV-XTD3 (6.16.16-windows-x86_64) to add photos to Domain Shared contacts, I receive an error stating: **Update Failed: [WinError 2] The system cannot find the file specified: 'company-logo.png'**

This could certainly be user error on my end, but I cannot find any documentation online that provides an example of how to use the "update contactphotos" function, and I've tried nearly every possible combination that I can think of. Here's an example command that is resulting in the error above:

`gam update contactphotos a80000609e74e5d sourcefolder "C:\GAMADV-XTD3" filename "company-logo.png"`

In addition, the following modified command:
`gam update contactphotos a80000609e74e5d filename "C:\GAMWork\company-logo.png"`

returns a different error:
**Domain: domain.com, Update maximum of 1 Photo
  Domain: domain.com, Contact: a80000609e74e5d, Photo:, Does not exist**

For reference, "company-logo.png" currently exists in "GAMADV-XTD3", "GAMConfig", "GAMWork", and even "drivedir", but the error is persistent.

The error appears to be related to python absolute path, and I noticed there was a similar error that was resolved in v5.02.04: **Fixed race condition on Windows involving locking `oauth2.txt.lock`.  ERROR: [WinError 2] The system cannot find the file specified: 'C:\\GAMConfig\\oauth2.txt.lock'**

Is this a syntax error on my end, or is this an issue that can be resolved? Thanks for such an excellent tool!

**Comment by taers232c** [comment]
You should be running GAM out of GAMWork and the file should be there.

I'm in California (PDT) and am available now vis Meet/Zoom or tomorrow.

Ross
--
***@***.***

**Comment by taers232c** [comment]
It's a bug, fix on the way. For now, run the command from GAMWork and omit sourcefolder "C:\GAMADV-XTD3"

Ross
--
***@***.***

**Comment by gkmachine** [comment]
Thanks for the prompt response! I've tried the same commands out of the GAMWork directory, however, the errors seem to persist. In testing, I've also noticed that the error is different if I reference an image that _isn't_ actually present. Here's an example that I just ran:
```
C:\GAMWork>gam update contactphotos 2b7fdd830d5838fa filename company-logo.png
Domain: domain.com, Update maximum of 1 Photo
  Domain: domain.com, Contact: 200fdd830d5838fa, Photo:, Does not exist

C:\GAMWork>gam update contactphotos 200fdd830d5838fa filename company-logo-NOT-PRESENT.png
Domain: domain.com, Update maximum of 1 Photo
  Domain: domain.com, Contact: 200fdd830d5838fa, Photo: company-logo-NOT-PRESENT.png, Update Failed: [WinError 2] The system cannot find the file specified: 'company-logo-NOT-PRESENT.png'
```
In the first attempt above (company-logo.png), the image exists in the GAMWork directory. In the second attempt above (company-logo-NOT-PRESENT.png), the image does not exist at all. It's odd that in the first attempt, the error references a photo without a filename, even though I'm clearly passing the filename.

Does this command support PNG's, or is there something else I may be doing wrong here? I'd be happy to have a meeting with you on Wednesday morning (Pacific time as well) - just let me know how we might arrange that.

Thanks again for your help!
